src/interface.jl

```diff
@@ -25,17 +25,22 @@ Returns a set of Can-types that the data type `T` exhibits.
 See also [`@assign`](@ref).
 """
 function traits(T::Assignable)
-    return get!(traits_map, T, Set{DataType}())
+    base = get!(traits_map, T) do; Set{DataType}() end
+    for (Tmap, s) in pairs(traits_map)
+        if T !== Tmap && T <: Tmap
+            union!(base, s)
+        end
+    end
+    base
```

Comment:
Can we add `return` in the front?  In generally, I like to follow Invenia's [Blue style guide](https://github.com/invenia/BlueStyle#method-definitions) as much as possible.

Reply:
done